core/src/main/scala/com/databricks/labs/remorph/generators/sql/ExpressionGenerator.scala

```diff
@@ -83,15 +84,21 @@ class ExpressionGenerator extends Generator[ir.Expression, String] {
   }
 
   private def jsonAccess(ctx: GeneratorContext, j: ir.JsonAccess): String = {
-    val path = j.path.map {
-      case ir.Id(name, false) => s".$name"
-      case ir.Id(name, true) if isAlphanum(name) => s".$name"
-      case ir.Id(name, true) => s"['$name']"
-      case ir.IntLiteral(value) => s"[$value]"
-      case ir.StringLiteral(value) => s"['$value']"
+    val path = jsonPath(j.path).mkString
+    val anchorPath = if (path.head == '.') ':' +: path.tail else path
+    s"${expression(ctx, j.json)}$anchorPath"
+  }
+
+  private def jsonPath(j: ir.Expression): Seq[String] = {
+    j match {
+//      case ir.Id(name, false) => Seq(s".$name")
+      case ir.Id(name, _) if isValidIdentifier(name) => Seq(s".$name")
+      case ir.Id(name, _) => Seq(s"['$name']".replace("'", "\""))
```

Comment:
Databricks does not access `'` single quotes that's why I have replaced it with `"` double quotes. This will not work - `demo.level_key:level_1_key.level_2_key['1']` but this works - `demo.level_key:level_1_key.level_2_key["1"]`

Reply:
done. added the test cases